test/feature-benchmark/mzcompose.py

```diff
@@ -131,7 +132,7 @@ def run_one_scenario(
         )
 
         if tag == "common-ancestor":
-            tag = benchmark_utils.resolve_tag_of_common_ancestor()
+            tag = docker.resolve_ancestor_image_tag_for_comparison()
```

Comment:
This means the machinery of `VersionList` is being bypassed and is not going to be used. This includes the `INVALID_VERSIONS` list in that file. Next time there is an invalid version that needs skipping, you will need to re-implement the list.

Please consider one or the other of the two approches:
- Remove the `VersionList` module if you are not going to use it going forward.
- Wire everything through `VersionList` and expand its API if needed to suit your needs.

As mentioned previously, we would want all tests to reach the same decision as to which versions to test. If a certain Platform checks scenario just takes the latest 4 versions from `version_list.py` and upgrades across them, while the feature benchmark performs complex logic around whether something is a release or not, or tagged or not, or in buildkite or not or in a PR or not, they are not going to be testing the same thing.

Reply:
I am checking `INVALID_VERSIONS` in https://github.com/MaterializeInc/materialize/pull/23145/files#diff-83c631245c55ffe8b757f79241dcc854baac54aabca6632467ef633c9886bbbcR299.
Furthermore, I have a follow-up PR which will check if an image exists on docker (https://github.com/MaterializeInc/materialize/pull/23183/commits/5b1bd3bfa915e0f935268b449aea773cf3d2247f but parked until this PR lands).

I intend to get rid of `VersionsFromDocs` (and similar) in a follow-up PR if we agree that this PR works as intended. See PR description:
>  (Adjusting platform upgrade checks and other checks using VersionsFromDocs requires some more efforts and will be done in a follow-up PR.)

Reply:
Though, it could be worth considering not immediately aligning both variants and first awaiting how this new method performs before replacing `VersionsFromDocs` with it. What do you think?

Reply:
No, feel free to challenge my solution!